src/components/Profile.js

```diff
@@ -121,30 +127,75 @@ function Profile() {
         </AllCollectionsWrapper>
       );
     } else if (showSection === MY_SCHEDULE) {
-      return <div>welcome to my schedule</div>;
+      return <div>welcom to my schedule</div>;
     }
   };
 
+  const getPins = async (id) => {
+    const querySnapshot = await getDocs(collection(db, "user", id, "pin"));
+    let myPins = [];
+    querySnapshot.forEach((doc) => {
+      myPins.push({...doc.data()});
+    });
+    setPins(myPins);
+  };
+
+  useEffect(() => {
+    userData && getPins(userData.id);
+  });
+
+  const getUserData = (userId) => {
+    const unsub = onSnapshot(doc(db, "user/" + userId), (doc) => {
+      if (!props.uid) {
+        return;
+      } else if (props.uid) {
```

Comment:
You do not need this judgment since you have already early returned.

Reply:
judgment removed